### PR TITLE
PivotGrid - tests for export of "Totals" settings

### PR DESCRIPTION
**Pull request by novsstation**
tests for export of "Totals" settings

### CODE REVIEW

File: testing/tests/DevExpress.exporter/exceljsParts/exceljs.pivotGrid.tests.js [hunk 1]
```diff
@@ -165,6 +165,7 @@ QUnit.module('Scenarios', moduleConfig, () => {
         const ds = {
             fields: [
                 { area: 'row', dataField: 'row1', dataType: 'string' },
+                { area: 'column', dataField: 'col1', dataType: 'string' },
```
Comment (IgnatovDan) [inline testing/tests/DevExpress.exporter/exceljsParts/exceljs.pivotGrid.tests.js]: imho we should not change existing tests in this way: there the [row: row1, data: count] config was tested and after this change this config is not tested.

File: testing/tests/DevExpress.exporter/exceljsParts/exceljs.pivotGrid.tests.js [hunk 2]
```diff
@@ -180,32 +181,35 @@ QUnit.module('Scenarios', moduleConfig, () => {
 
         const expectedCells = [[
             { excelCell: { value: '', alignment: alignCenterTopWrap }, pivotCell: { alignment: 'left', colspan: 1, rowspan: 1, text: '', width: 100 } },
-            { excelCell: { value: 'Grand Total', alignment: alignCenterTopWrap }, pivotCell: { area: 'column', colspan: 1, isLast: true, rowspan: 1, text: 'Grand Total', type: 'GT', width: 100 } }
+            { excelCell: { value: 'a', alignment: alignCenterTopWrap }, pivotCell: { area: 'column', colspan: 1, rowspan: 1, isLast: true, path: ['a'], text: 'a', type: 'D', dataSourceIndex: 1, width: 100 } },
```
Comment (IgnatovDan) [inline testing/tests/DevExpress.exporter/exceljsParts/exceljs.pivotGrid.tests.js]: imho this is a significant change in the initial config so the initial config is not tested with these changes

File: testing/tests/DevExpress.exporter/exceljsParts/exceljs.pivotGrid.tests.js [hunk 3]
```diff
@@ -180,32 +181,35 @@ QUnit.module('Scenarios', moduleConfig, () => {
...
             done();
         });
     });
 
-    QUnit.test('Export [string x string x number] with row grand totals', function(assert) {
+    QUnit.test('Export [string x string x number] & showRowGrandTotals', function(assert) {
         const done = assert.async();
         const ds = {
             fields: [
+                { area: 'row', dataField: 'row1', dataType: 'string' },
```
Comment (IgnatovDan) [inline testing/tests/DevExpress.exporter/exceljsParts/exceljs.pivotGrid.tests.js]: again significant change in the tested config: the old config is not tested with new changes

File: testing/tests/DevExpress.exporter/exceljsParts/exceljs.pivotGrid.tests.js [hunk 4]
```diff
@@ -222,24 +226,76 @@ QUnit.module('Scenarios', moduleConfig, () => {
...
             helper.checkMergeCells(expectedCells, topLeft);
             helper.checkOutlineLevel([0, 0], topLeft.row);
             helper.checkAutoFilter(false, { from: topLeft, to: topLeft }, { state: 'frozen', ySplit: topLeft.row, xSplit: topLeft.column });
-            helper.checkCellRange(cellRange, { row: 2, column: 2 }, topLeft);
+            helper.checkCellRange(cellRange, { row: 3, column: 2 }, topLeft);
+            done();
+        });
+    });
+
+    QUnit.test('Export [string x string x number] & showRowGrandTotals & showRowGrandTotals', function(assert) {
```
Comment (IgnatovDan) [inline testing/tests/DevExpress.exporter/exceljsParts/exceljs.pivotGrid.tests.js]: & showRowGrandTotals & showRowGrandTotals - show row grand totals twice?

File: testing/tests/DevExpress.exporter/exceljsParts/exceljs.pivotGrid.tests.js [hunk 5]
```diff
@@ -869,70 +925,67 @@ QUnit.module('Scenarios', moduleConfig, () => {
         });
     });
 
-    QUnit.test('Export [string(A,B) x None x number]', function(assert) {
+    QUnit.test('Export [string(A,B) x None x None] & showRowGrandTotals', function(assert) {
         const done = assert.async();
         const ds = {
             fields: [
                 { area: 'row', dataField: 'row1' },
-                { area: 'data', summaryType: 'count', dataType: 'number' }
```
Comment (IgnatovDan) [inline testing/tests/DevExpress.exporter/exceljsParts/exceljs.pivotGrid.tests.js:877]: The initial config was changed significantly: now, there is no 'data' area options and config details are not obvious with new changes.

File: testing/tests/DevExpress.exporter/exceljsParts/exceljs.pivotGrid.tests.js [hunk 6]
```diff
@@ -869,70 +925,67 @@ QUnit.module('Scenarios', moduleConfig, () => {
...
-                { row1: 'B' },
-                { row1: 'B' }
+                { row1: 'A' }, { row1: 'B' }
             ]
         };
 
         const pivotGrid = $('#pivotGrid').dxPivotGrid({
             showColumnGrandTotals: false,
-            showRowGrandTotals: false,
+            showRowGrandTotals: true,
```
Comment (IgnatovDan) [inline testing/tests/DevExpress.exporter/exceljsParts/exceljs.pivotGrid.tests.js]: again significant change in config: the old config is not tested with new changes

File: testing/tests/DevExpress.exporter/exceljsParts/exceljs.pivotGrid.tests.js [hunk 7]
```diff
@@ -869,70 +925,67 @@ QUnit.module('Scenarios', moduleConfig, () => {
...
         });
     });
 
-    QUnit.test('Export [string(A,B) x None x number] & showColumnGrandTotals', function(assert) {
+    QUnit.test('Export [string(A,B) x None x None] & showRowGrandTotals & showColumnGrandTotals', function(assert) {
         const done = assert.async();
         const ds = {
             fields: [
                 { area: 'row', dataField: 'row1' },
-                { area: 'data', summaryType: 'count', dataType: 'number' }
```
Comment (IgnatovDan) [inline testing/tests/DevExpress.exporter/exceljsParts/exceljs.pivotGrid.tests.js:924]: Unclear change: "area: data" is a part of the tested config.
Why was this line removed?

File: testing/tests/DevExpress.exporter/exceljsParts/exceljs.pivotGrid.tests.js [hunk 8]
```diff
@@ -941,37 +994,42 @@ QUnit.module('Scenarios', moduleConfig, () => {
...
     });
 
-    QUnit.test('Export [string(A,B) x string x None]', function(assert) {
+    QUnit.test('Export [string(A,B) x None x number]', function(assert) {
         const done = assert.async();
         const ds = {
             fields: [
                 { area: 'row', dataField: 'row1' },
-                { area: 'column', dataField: 'col1' },
+                { area: 'data', summaryType: 'count', dataType: 'number' }
```
Comment (IgnatovDan) [inline testing/tests/DevExpress.exporter/exceljsParts/exceljs.pivotGrid.tests.js]: Again significant change: the old config is not tested with new changes.

**Review by IgnatovDan — CHANGES_REQUESTED** [review]

File: testing/tests/DevExpress.exporter/exceljsParts/exceljs.pivotGrid.tests.js (hunk 3 above)
Comment (novsstation) [reply testing/tests/DevExpress.exporter/exceljsParts/exceljs.pivotGrid.tests.js]: Previously, there is fictitious configuration, not synchronized with description. 
Previous configurations  reproduce the empty grid with "no data".

Test description is [string x string x number], what means row x col x data. 
And according to this description there must be data.

It was a simple typo, because in other test with  [string x string x number] everything is ok

File: testing/tests/DevExpress.exporter/exceljsParts/exceljs.pivotGrid.tests.js (hunk 2 above)
Comment (novsstation) [reply testing/tests/DevExpress.exporter/exceljsParts/exceljs.pivotGrid.tests.js]: Previously, there is fictitious configuration, not synchronized with description. 
Previous configurations  reproduce the empty grid with "no data".

Test description is [string x string x number], what means row x col x data. 
And according to this description there must be data.

It was a simple typo, because in other test with  [string x string x number] everything is ok

File: testing/tests/DevExpress.exporter/exceljsParts/exceljs.pivotGrid.tests.js (hunk 4 above)
Comment (novsstation) [reply testing/tests/DevExpress.exporter/exceljsParts/exceljs.pivotGrid.tests.js]: Fixed

**Review by IgnatovDan — APPROVED** [review]

**Review by jsdmitry — APPROVED** [review]
